test/hotspot/jtreg/compiler/vectorization/TestOptionVectorizeIR.java

```diff
@@ -0,0 +1,201 @@
+    @Run(test = "test3")
+    @Warmup(100)
+    public void runTest3() {
+        int[] data = new int[RANGE];
+        test1(data);
+        test3(data, 2, 3);
+        verify("test3", data, gold3);
+    }
+
+    @Run(test = "test4")
+    @Warmup(100)
+    public void runTest4() {
+        int[] data = new int[RANGE];
+        test1(data);
+        test4(data);
+        verify("test4", data, gold4);
+    }
+
+    @Run(test = "test5")
+    @Warmup(100)
+    public void runTest5() {
+        int[] data = new int[RANGE];
+        test1(data);
+        test5(data);
+        verify("test5", data, gold5);
+    }
+
+    @Run(test = "test6")
+    @Warmup(100)
+    public void runTest6() {
+        int[] data = new int[RANGE];
+        test1(data);
+        test6(data);
+        verify("test6", data, gold6);
+    }
+
+    @Test
+    @IR(applyIfCPUFeatureOr = {"avx2", "true", "sve", "true"}, counts = {IRNode.POPULATE_INDEX, "> 0"})
+    @IR(applyIfCPUFeatureOr = {"avx2", "true", "sve", "true"}, counts = {IRNode.STORE_VECTOR, "> 0"})
+    static void test1(int[] data) {
+       for (int j = 0; j < RANGE; j++) {
+           // Vectorizes even if it is not forced
+           data[j] = j;
+       }
+    }
+
+    @Test
+    @IR(counts = {IRNode.LOAD_VECTOR, "> 0"})
+    @IR(counts = {IRNode.ADD_VI, "> 0"})
+    @IR(counts = {IRNode.STORE_VECTOR, "> 0"})
+    static void test2(int[] data) {
+       for (int j = 0; j < RANGE - 1; j++) {
+           // Only vectorizes if forced, because of offset by 1
+           data[j] = data[j] + data[j + 1];
+       }
+    }
+
+    @Test
+    @IR(counts = {IRNode.LOAD_VECTOR, "> 0"})
+    @IR(counts = {IRNode.REPLICATE_I, "> 0"})
+    @IR(counts = {IRNode.ADD_VI, "> 0"})
+    @IR(counts = {IRNode.MUL_VI, "> 0"})
+    @IR(counts = {IRNode.STORE_VECTOR, "> 0"})
+    static void test3(int[] data, int A, int B) {
+       for (int j = 0; j < RANGE - 1; j++) {
+           // Only vectorizes if forced, because of offset by 1
+           data[j] = A * data[j] + B * data[j + 1];
+       }
+    }
+
+    @Test
+    @IR(counts = {IRNode.LOAD_VECTOR, "= 0"})
+    @IR(counts = {IRNode.STORE_VECTOR, "= 0"})
+    static void test4(int[] data) {
+       for (int j = 0; j < RANGE - 1; j++) {
+           // write forward -> cyclic dependency -> cannot vectorize
+           // independent(s1, s2) for adjacent loads should detect this
+           data[j + 1] = data[j];
+       }
+    }
+
+    @Test
+    @IR(counts = {IRNode.LOAD_VECTOR, "= 0"})
+    @IR(counts = {IRNode.STORE_VECTOR, "= 0"})
+    static void test5(int[] data) {
+       for (int j = 0; j < RANGE - 3; j++) {
+           // write forward -> cyclic dependency -> cannot vectorize
+           // independent(s1, s2) for adjacent loads cannot detect this
+           // Checks with memory_alignment are disabled via compile option
+           data[j + 2] = data[j];
+       }
+    }
+
+    @Test
+    @IR(counts = {IRNode.LOAD_VECTOR, "= 0"})
+    @IR(counts = {IRNode.STORE_VECTOR, "= 0"})
```

Comment:
All of these (also in other tests) should be replaced by something like:

```suggestion
    @IR(failOn = {IRNode.LOAD_VECTOR, IRNode.STORE_VECTOR})
```